hypothesis-python/tests/cover/test_settings.py

```diff
@@ -324,6 +324,54 @@ def test_setattr_on_settings_singleton_is_error():
         settings.max_examples = 10
 
 
+def test_deadline_given_valid_int():
+    x = settings(deadline=1000).deadline
+    assert isinstance(x, datetime.timedelta)
+    assert x.days == 0 and x.seconds == 1 and x.microseconds == 0
+
+
+def test_deadline_given_zero():
```

Comment:
All of the tests for invalid deadlines can be consolidated as follows:

```python
@pytest.mark.parametrize("x", [0, -0.7, 86400000000000000.2, ...])
def test_invalid_deadline(x):
    with pytest.raises(InvalidArgument):
        settings(deadline=0)
```

I'm really impressed by the comprehensiveness of the tests too - good work!